test/test_yaml_utils.py

```diff
@@ -264,6 +266,38 @@ def test_formatted_yaml_loader_dumper(
     assert not list(run_yamllint(after_content, config))
 
 
+def test_formatted_task() -> None:
+    """Ensure that FormattedYAML loads/dumps formatting fixtures consistently.
+
+    Compares the data of task file before and after formatting,
+    to ensure that the single space between blocks is preserved.
+    """
+    task_file_path = formatting_tasks_dir / "task-fmt.yml"
+    task_file_content = task_file_path.read_text()
+
+    yaml = ansiblelint.yaml_utils.FormattedYAML()
+
+    before_formatting = yaml.loads(task_file_content)
+    dump_from_formatting = yaml.dumps(before_formatting)
+    assert task_file_content == dump_from_formatting
```

Comment:
Another question that I have: if I comment the code change and only include the tests, are they going to pass? If they pass already it means they have only very limited value. If not, they are likely ok.

Reply:
I see that the body of the test looks identical. Maybe it would be better to use parameterized tests instead of repeating the same body of the test. Also this should make it easier to add other test files later, for other test-cases.

Reply:
> Another question that I have: if I comment the code change and only include the tests, are they going to pass? If they pass already it means they have only very limited value. If not, they are likely ok.

Yes, these tests still passes even if I comment the code change. Because of the fact that single space between tasks is being preserved, the task file mentioned in this [issue](https://github.com/ansible/ansible-lint/issues/3603) was one of the unusual scenario where it fails. And the code I added is only to support that.